apps/api/src/hyp3_api/validation.py

```diff
@@ -136,6 +140,45 @@ def get_multipolygon_from_geojson(input_file):
     return MultiPolygon(polygons)
 
 
+def check_bounds_formatting(job, _):
+    bounds = job['job_parameters']['bounds']
+    if bounds == [0.0, 0.0, 0.0, 0.0]:
+        return
+
+    if bounds[0] >= bounds[2] or bounds[1] >= bounds[3]:
+        raise BoundsValidationError(
+            'Invalid order for bounds. Bounds should be ordered [min lon, min lat, max lon, max lat].'
+        )
+
+    def bad_lat(lat):
+        return lat > 90 or lat < -90
+
+    def bad_lon(lon):
+        return lon > 180 or lon < -180
+
+    if any([bad_lon(bounds[0]), bad_lon(bounds[2]), bad_lat(bounds[1]), bad_lat(bounds[3])]):
+        raise BoundsValidationError(
+            'Invalid lon/lat value(s) in bounds. Bounds should be ordered [min lon, min lat, max lon, max lat].'
+        )
+
+
+def check_granules_intersecting_bounds(job, granule_metadata):
+    bounds = job['job_parameters']['bounds']
+    if bounds == [0.0, 0.0, 0.0, 0.0]:
+        bounds = granule_metadata[0]['polygon']
```

Comment:
Is the granule order always preserved in the CMR search response?

Reply:
Couldn't find the answer in the CMR docs, but some quick local testing suggests yes.